orchagent/portsorch.cpp

```diff
@@ -3999,6 +4003,490 @@ void PortsOrch::doLagMemberTask(Consumer &consumer)
+
+    return task_success;
+}
+
+task_process_status PortsOrch::deleteVlanXlateEntry(const VlanXlateKey& key)
+{
+    auto it = m_vlan_xlate->vlan_xlate_map.find(key);
+    if (it != m_vlan_xlate->vlan_xlate_map.end())
+    {
+        auto status = sai_vlan_api->remove_vlan_stack(it->second.ingress_entry);
+        if (status != SAI_STATUS_SUCCESS) {
+            SWSS_LOG_ERROR("Delete VLAN xlate entry ingress failed, port:%" PRIx64 ", s_vlanid:%d", key.port_oid, key.s_vlanid);
+            return handleSaiRemoveStatus(SAI_API_VLAN, status);
+        }
+
+        status = sai_vlan_api->remove_vlan_stack(it->second.egress_entry);
+        if (status != SAI_STATUS_SUCCESS) {
+            SWSS_LOG_ERROR("Delete VLAN xlate entry egress failed, port:%" PRIx64 ", s_vlanid:%d", key.port_oid, key.s_vlanid);
+            return handleSaiRemoveStatus(SAI_API_VLAN, status);
+        }
+
+        m_vlan_xlate->vlan_xlate_map.erase(key);
+    }
+
+    return task_success;
+}
+
+task_process_status PortsOrch::doVlanStackTask(const string& key, const string& op, const vector<swss::FieldValueTuple>& values)
+{
+    task_process_status task_status;
+    vector<string> keys = tokenize(key, ':');
+    Port port;
+
+    if (!getPort(keys.at(0), port))
+    {
+        return task_need_retry;
+    }
+
+    if (port.m_type != Port::PHY && port.m_type != Port::LAG)
+    {
+        SWSS_LOG_ERROR("VlanStack failed due to incorrect port type (%d).", port.m_type);
+        return task_failed;
+    }
+
+    if (op == SET_COMMAND)
+    {
+        string action, s_vlanid;
+        uint8_t s_vlan_priority = 0;
+        vector<uint16_t> c_vlanids;
+
+        s_vlanid = keys.at(1);
+        if (!isValidVlanId(s_vlanid, 1, 4094))
+        {
+            return task_failed;
+        }
+
+        for (auto itp : values)
+        {
+            string attr_name = fvField(itp);
+            string attr_value = to_upper(fvValue(itp));
+
+            if (attr_name == "c_vlanids")
+            {
+                auto rv = parseVlanList(attr_value, c_vlanids);
+                if (rv == false)
+                {
+                    return task_failed;
+                }
+            }
+            else if (attr_name == "s_vlan_priority")
+            {
+                try
+                {
+                    s_vlan_priority = (uint8_t)stoi(attr_value);
+                }
+                catch (std::invalid_argument& e)
+                {
+                    return task_failed;
+                }
+
+                if (s_vlan_priority < 0 || s_vlan_priority > 7)
+                {
+                    return task_failed;
+                }
+            }
+        }
+
+        sai_object_id_t port_oid = port.m_type == Port::PHY ? port.m_port_id : port.m_lag_id;
+        VlanStackKey stack_key = { port_oid, (uint16_t)to_uint<uint16_t>(s_vlanid) };
+
+        task_status = deleteVlanStackEntry(stack_key);
```

Comment:
Why a delete is issued before Adding the Stacking Entry?

Reply:
To make modify action easy, so use remove and add to support update